---
title: "Analysis of RNA sequencing data from Chen et. al. examining PDXs that are sensitive or resistant to HIF2a antagonists"
author: "Katja Lamia"
date: "2/17/2024"
output:
  pdf_document: default
  html_document: default
---

```{r setup, echo=FALSE, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, include=FALSE)
```

```{r load packages}
library(ggpubr)
library(rstatix)
library(tidyverse)
library(dplyr)
library(ggplot2)
library(ggrepel)

```

Rebecca downloaded the RNA sequencing data from the paper Chen et al. and aligned them to the mouse transcriptome. Katja ran DESEQ2 through GenePattern with sensitive vs resistant samples as the main factor and vehicle vs PT treatment as a confounding factor.

```{r read in data for vehicle vs PT2399 DESEQ2 output for sensitive PDX samples only}

#read in data generated by DESeq2 from Chen et al RNAseq data comparing vehicle vs PT2399 in sensitive PDXs

DEGsPT2399_sensitive <-
  read_delim("Chen2016RNAseq/Chen_SensitiveONLY_Ctrl_vs_PT.PT.vs.Veh.DESeq2_results_report.txt") %>% 
  select(id,log2FoldChange,padj,pvalue) %>%
  rename("Log2FC" = "log2FoldChange")

DEGsPT2399_sensitive$diffexp <- "NO"

#Set thresholds
ThFC <- 1.5
ThSig <- 0.05
Include <- 0 #If you want to reduce the file sizes and run times by excluding points with very little expression change between groups, can increase this number.

#Reduce number of points
DEGsPT2399_sensitivePlot <- DEGsPT2399_sensitive %>% filter(Log2FC > log2(1 + Include) | 
                              Log2FC < log2(1 - Include)
                            )

#Define UP and DOWN DEGsPT2399_sensitive
DEGsPT2399_sensitivePlot$diffexp[DEGsPT2399_sensitivePlot$Log2FC > log2(ThFC) & 
                   DEGsPT2399_sensitivePlot$padj < ThSig] <- "UP"
DEGsPT2399_sensitivePlot$diffexp[DEGsPT2399_sensitivePlot$Log2FC < log2(1/ThFC) & 
                   DEGsPT2399_sensitivePlot$padj < ThSig] <- "DOWN"

#Add column for labels
DEGsPT2399_sensitivePlot$delabel <- NA

#Define genes to label - default label top 10 genes. Can change this number if desired.
N <- 0      # Don't label any point
#N <- 1368   # Label all point
#N <- 100    # Label N points.
TopDEGsPT2399_sensitive <- head(arrange(DEGsPT2399_sensitivePlot, pvalue), N)
Pthres = TopDEGsPT2399_sensitive$pvalue[N]

DEGsPT2399_sensitivePlot$delabel[DEGsPT2399_sensitivePlot$pvalue <= Pthres] <- DEGsPT2399_sensitivePlot$id[DEGsPT2399_sensitivePlot$pvalue <= Pthres]

vplot <- ggplot(data = DEGsPT2399_sensitivePlot, 
       aes(x = Log2FC, 
           y = -log10(padj), 
           col = diffexp, 
           label = delabel)) + 
  geom_point() + 
  geom_text_repel() +
  theme_minimal() +
  scale_color_manual(values = c("blue", "black", "red")) +
  theme(text = element_text(size = 10))

 
#Add vertical and horizontal lines to plot
vplot +   ##xlim(-10,10) +   #ylim(0,12) +
  geom_vline(xintercept = c(log2(ThFC), log2(1/ThFC)), col = 'gray') +
  geom_hline(yintercept = -log10(ThSig), col = 'gray')

```


```{r define groups of genes that are dependent on ARNT and/or BMAL1 in 786O cells}

# Read results from DESeq2 for each shRNA

Sig786O_ARNT <- read.csv("MelloRNAseq/786O_shARNT_results.csv") %>% rename(id = X)
Sig786O_BMAL1 <- read.csv("MelloRNAseq/786O_shBMAL1_results.csv") %>% rename(id = X)

# Identify transcripts that are significantly reduced by shRNA for ARNT and/or BMAL1
Sig786O_shARNT_DN <- Sig786O_ARNT %>% filter(log2FoldChange < 0)
Sig786O_shBMAL1_DN <- Sig786O_BMAL1 %>% filter(log2FoldChange < 0)

Sig786O_shARNTsp_DN <- Sig786O_shARNT_DN %>% filter(!(id %in% Sig786O_shBMAL1_DN$id))
write_excel_csv(Sig786O_shARNTsp_DN, "ARNTsp_DN_786O.xls")
Sig786O_shBMAL1sp_DN <- Sig786O_shBMAL1_DN %>% filter(!(id %in% Sig786O_shARNT_DN$id))
write_excel_csv(Sig786O_shBMAL1sp_DN, "BMAL1sp_DN_786O.xls")
Sig786O_shAandB1_DN <- Sig786O_shARNT_DN %>% filter(id %in% Sig786O_shBMAL1_DN$id)
write_excel_csv(Sig786O_shAandB1_DN, "AandB1_DN_786O.xls")


```


```{r volcano plots for 786O bHLH-sp genes reflecting Log2FC for PT2399 treatment in sensitive PDXs}

DEGsPT2399_sensitivePlot$bHLHact <- "NA"

# Define genes that are uniquely reduced by each shRNA in 786O cells
DEGsPT2399_sensitivePlot$bHLHact <- "NA"
DEGsPT2399_sensitivePlot$bHLHact[DEGsPT2399_sensitivePlot$id %in% Sig786O_shARNTsp_DN$id] <- "ARNT"
DEGsPT2399_sensitivePlot$bHLHact[DEGsPT2399_sensitivePlot$id %in% Sig786O_shBMAL1sp_DN$id] <- "BMAL1"

#Define overlapping targets
DEGsPT2399_sensitivePlot$bHLHact[DEGsPT2399_sensitivePlot$id %in% Sig786O_shAandB1_DN$id] <- "A_and_B1"



vplot <- ggplot(data = DEGsPT2399_sensitivePlot %>% filter(bHLHact == "ARNT"), 
       aes(x = Log2FC, 
           y = -log10(padj), 
           col = diffexp, 
           label = delabel)) + 
  geom_point() + 
  geom_text_repel() +
  theme_minimal() +
  scale_color_manual(values = c("blue", "black", "red")) +
  labs(title = "ARNTsp-activated (786O)") +
  theme(text = element_text(size = 10))
vplot +   
  ##xlim(-10,10) +   
  #ylim(0,12) +
  geom_vline(xintercept = c(log2(ThFC), log2(1/ThFC)), col = 'gray') +
  geom_hline(yintercept = -log10(ThSig), col = 'gray')
ggsave("FIG3J.png", height = 3, width = 4, units = "in")

vplot <- ggplot(data = DEGsPT2399_sensitivePlot %>% filter(bHLHact == "BMAL1"), 
       aes(x = Log2FC, 
           y = -log10(padj), 
           col = diffexp, 
           label = delabel)) + 
  geom_point() + 
  geom_text_repel() +
  theme_minimal() +
  scale_color_manual(values = c("blue", "black", "red")) +
  labs(title = "BMAL1sp-activated (786O)") +
  theme(text = element_text(size = 10))
vplot +
  ##xlim(-10,10) +
  #ylim(0,12) +
  geom_vline(xintercept = c(log2(ThFC), log2(1/ThFC)), col = 'gray') +
  geom_hline(yintercept = -log10(ThSig), col = 'gray')
ggsave("FIG3K.png", height = 3, width = 4, units = "in")

vplot <- ggplot(data = DEGsPT2399_sensitivePlot %>% filter(bHLHact == "A_and_B1"), 
       aes(x = Log2FC, 
           y = -log10(padj), 
           col = diffexp, 
           label = delabel)) + 
  geom_point() + 
  geom_text_repel() +
  theme_minimal() +
  scale_color_manual(values = c("blue", "black", "red")) +
  labs(title = "ARNT and BMAL1 activated (786O)") +
  theme(text = element_text(size = 10))
vplot +
  ##xlim(-10,10) +  
  #ylim(0,12) +
  geom_vline(xintercept = c(log2(ThFC), log2(1/ThFC)), col = 'gray') +
  geom_hline(yintercept = -log10(ThSig), col = 'gray')
ggsave("FIG3L.png", height = 3, width = 4, units = "in")

vplot <- ggplot(data = DEGsPT2399_sensitivePlot %>% filter(bHLHact == "NA"), 
       aes(x = Log2FC, 
           y = -log10(padj), 
           col = diffexp, 
           label = delabel)) + 
  geom_point() + 
  geom_text_repel() +
  theme_minimal() +
  scale_color_manual(values = c("blue", "black", "red")) +
  labs(title = "Other") +
  theme(text = element_text(size = 10))
vplot +
  ##xlim(-10,10) +  
  #ylim(0,12) +
  geom_vline(xintercept = c(log2(ThFC), log2(1/ThFC)), col = 'gray') +
  geom_hline(yintercept = -log10(ThSig), col = 'gray')
ggsave("Other786O_PT2399impact_sensitive.png", height = 3, width = 4, units = "in")


# Create boxplots

DEGsPT2399_sensitivePlotGrouped <- DEGsPT2399_sensitivePlot %>%
  group_by(bHLHact)

DEGsPT2399_sensitive_TwoWayTukey_bHLH_786O <- aov(Log2FC ~ bHLHact,
                         data = DEGsPT2399_sensitivePlotGrouped) %>%
  tukey_hsd() %>% add_significance()

library(forcats)

colors <- c("gray", "yellow", "purple", "salmon")

PlotData <- DEGsPT2399_sensitivePlot %>%
  mutate(bHLHact = fct_relevel(bHLHact, 
            "NA", "ARNT", "BMAL1", "A_and_B1"))

ptest <- ggplot(PlotData %>% filter(padj < 0.05), aes(x = bHLHact, y = Log2FC, fill = bHLHact))
ptest + geom_boxplot() + scale_fill_manual(values = colors) + theme_classic() +
  labs(title = "Impact of PT2399 in sensitive PDXs on transcripts activated by ARNT and/or BMAL1 (786O)") +
  theme(text = element_text(size = 10))
ggsave("FIG3I.png", height = 3, width = 4, units = "in")



```
